### PR TITLE
Add support for marketplace plan information

### PR DESCRIPTION
**Pull request by jasonwbarnett**
### Description

Adds support for marketplace images that have associated plans.

I manually ran integration tests and this works as expected. With and without the plan information.

### Issues Resolved

N/A

### Check List

- [x] New functionality includes tests
- [x] All tests pass
- [X] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

**Comment by jasonwbarnett** [comment]
Ready for review

File: lib/kitchen/driver/azurerm.rb [hunk 1]
```diff
@@ -20,6 +20,8 @@ class Azurerm < Kitchen::Driver::Base
       attr_accessor :resource_management_client
       attr_accessor :network_management_client
 
+      kitchen_driver_api_version 2
```
Comment (jasonwbarnett) [inline lib/kitchen/driver/azurerm.rb:23]: I mainly added this cause I saw across the board the primary drivers have this set. Willing to pull if this is a big deal.

**Comment by jasonwbarnett** [comment]
Ready for review

**Comment by jasonwbarnett** [comment]
Ready for review

**Comment by tas50** [comment]
Looks good. Thanks @jasonwbarnett

**Pull request merged**